tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -172,10 +172,69 @@ public void testNoModuleInTOML() throws BallerinaTestException {
+        bMainInstance.runMain("run", new String[]{"main"}, addSecretEnvVariable(secretFilePath), new String[]{},
+                              new LogLeecher[]{runLeecher}, projectPath.toString());
+        runLeecher.waitForText(5000);
+    }
+
+    @Test
+    public void testEncryptedConfigsWithIncorrectSecret() throws BallerinaTestException {
+        Path projectPath = Paths.get(testFileLocation, "encryptedConfigProject").toAbsolutePath();
+        String secretFilePath = Paths.get(testFileLocation, "Secrets", "incorrectSecret.txt").toString();
+        LogLeecher runLeecher = new LogLeecher("error: failed to retrieve the encrypted value for variable: " +
+                                                       "'password' : Given final block not properly padded. Such " +
+                                                       "issues can arise if a bad key is used during decryption.",
+                                               LogLeecher.LeecherType.ERROR);
+        bMainInstance.runMain("run", new String[]{"main"}, addSecretEnvVariable(secretFilePath), new String[]{},
+                              new LogLeecher[]{runLeecher}, projectPath.toString());
+        runLeecher.waitForText(5000);
+    }
+
+    @Test
+    public void testEncryptedConfigsWithEmptySecret() throws BallerinaTestException {
+        Path projectPath = Paths.get(testFileLocation, "encryptedConfigProject").toAbsolutePath();
+        String secretFilePath = Paths.get(testFileLocation, "Secrets", "emptySecret.txt").toString();
+        LogLeecher runLeecher = new LogLeecher("error: failed to initialize the cipher tool due to empty secret text",
+                                               LogLeecher.LeecherType.ERROR);
+        bMainInstance.runMain("run", new String[]{"main"}, addSecretEnvVariable(secretFilePath), new String[]{},
+                              new LogLeecher[]{runLeecher}, projectPath.toString());
+        runLeecher.waitForText(5000);
+    }
+
+    @Test
+    public void testInvalidAccessEncryptedConfigs() throws BallerinaTestException {
+        Path projectPath = Paths.get(testFileLocation, "encryptedConfigProject").toAbsolutePath();
+        String configFilePath = Paths.get(testFileLocation, "ConfigFiles", "InvalidEncryptedConfig.toml").toString();
+        String secretFilePath = Paths.get(testFileLocation, "Secrets", "correctSecret.txt").toString();
+        LogLeecher runLeecher = new LogLeecher("error: failed to retrieve the encrypted value for variable: " +
+                                                       "'password' : Input byte array has wrong 4-byte ending unit",
+                                               LogLeecher.LeecherType.ERROR);
+        bMainInstance.runMain("run", new String[]{"main"}, addEnvVariables(configFilePath, secretFilePath),
+                              new String[]{},
+                              new LogLeecher[]{runLeecher}, projectPath.toString());
+        runLeecher.waitForText(5000);
```

Comment:
The code inside the test cases are repeating. We can move the method bodies to a single method and pass the required data as parameters.

Reply:
This happens in the other test cases as well. Shall I send a separate PR after the merge?

Reply:
I have used the same approach in all configurable test caces. :D +1 to fix all with a separate PR.